packages/shared/src/components/CreateMyFeedButton.tsx

```diff
@@ -0,0 +1,129 @@
+import React, { ReactElement, useContext, useEffect } from 'react';
+import classNames from 'classnames';
+import { IFlags } from 'flagsmith';
+import PlusIcon from '../../icons/plus.svg';
+import UserIcon from '../../icons/user.svg';
+import { Button } from './buttons/Button';
+import { Features, getFeatureValue } from '../lib/featureManagement';
+import AnalyticsContext from '../contexts/AnalyticsContext';
+import { AnalyticsEvent } from '../hooks/analytics/useAnalyticsQueue';
+import { getThemeColor, ThemeColor } from './utilities';
+
+interface CreateMyFeedButtonProps {
+  type: string;
+  sidebarExpanded?: boolean;
+  flags: IFlags;
+  action: () => unknown;
+}
+
+const getAnalyticsEvent = (
+  eventName: string,
+  copy: string,
+  type: string,
+): Partial<AnalyticsEvent> => ({
+  event_name: eventName,
+  target_type: 'my feed button',
+  target_id: type,
+  feed_item_title: copy,
+});
+
+interface ClassProps {
+  explainerColor?: ThemeColor;
+  sidebarExpanded?: boolean;
+}
+const wrapperClasses = (sidebarExpanded: ClassProps) => {
+  return {
+    sidebar: sidebarExpanded ? 'h-[8.125rem] justify-center' : 'h-[8.125rem] ',
+    feed_top: 'w-full items-center mb-8',
+  };
+};
+
+const innerWrapClasses = ({ explainerColor, sidebarExpanded }: ClassProps) => {
+  return {
+    sidebar: sidebarExpanded
+      ? `flex-col ${explainerColor.shadow} border p-3 m-4`
+      : 'flex-col mx-3',
+    feed_title: `flex-row-reverse`,
+    feed_top: `${explainerColor.shadow} p-2 border`,
+    feed_ad: `${explainerColor.shadow} p-2 border flex-col flex-1 justify-center`,
+  };
+};
+const textClass = (sidebarExpanded: ClassProps) => {
+  return {
+    sidebar: sidebarExpanded
+      ? 'typo-footnote w-[11.25rem] mb-3 transform opacity-100 ease-linear duration-200  delay-200'
+      : 'typo-footnote w-[11.25rem] mb-3 transform duration-0 delay-0 opacity-0',
```

Comment:
I think we can unify some of the classes.
```suggestion
    sidebar: classNames('typo-footnote w-[11.25rem] mb-3 transform', sidebarExpanded
      ? 'opacity-100 ease-linear duration-200  delay-200'
      : 'duration-0 delay-0 opacity-0'),
```